src/chat/resourceConnection.ts

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type TokenCredential } from "@azure/core-auth";
+import { type AzureSubscriptionProvider } from "@microsoft/vscode-azext-azureauth";
+
+export type AzureResourceConnection<TypeT = string, KindT = string | undefined> = {
+    type: TypeT;
+    kind?: KindT;
+    name: string;
+    subscriptionId: string;
+    resourceGroup: string;
+};
+
+export function partialConnectionHasAtLeastOneField(partialConnection: object): boolean {
```

Comment:
A bunch of these helpers probably aren't needed? Will need to double check next week while Jason is gone.